remill/Arch/AArch64/Arch.cpp

```diff
@@ -719,6 +719,28 @@ static bool DecodeBitMasks(uint64_t N /* one bit */,
   }
   return true;
 }
+/// Utility function for extracting [From, To] bits from a uint32_t.
+static inline unsigned slice(uint32_t Bits, unsigned From, unsigned To) {
+  assert(From < 32 && To < 32 && From >= To);
+  return (Bits >> To) & ((1 << (From - To + 1)) - 1);
+}
+
+// Adapted from LLVM 
+// https://llvm.org/svn/llvm-project/llvm/branches/ggreif/CallInst-operands/lib/Target/ARM/Disassembler/ARMDisassemblerCore.cpp
+static uint64_t VFPExpandImm(uint8_t imm, RegClass float_class) {
+  assert(float_class == kRegS || float_class == kRegD || float_class == kRegH);
+  uint64_t result;
+
+  unsigned bit6 = slice(imm, 6, 6);
+  if(kRegD == float_class) {
+    result = (uint64_t)slice(imm, 7, 7) << 63 | (uint64_t)slice(imm, 5, 0) << 48;
+    bit6? result |= 0xffULL << 54l : result |= 0x1ULL << 62;
```

Comment:
Don't do assignments inside of a ternary operator, just split this out into an `if` statement.

Also I suggest upper case `L` for the `54` shift constant, it kind of looks like `541` in some fonts. You probably want `UL` to, otherwise it's a signed `long`.

Reply:
Alternatively, instead of doing this:
```c
bit6 ? result |= 0xffULL << 54l : result |= 0x1ULL << 62;
```
do this:
```c
result |= bit6 ? (0xffULL << 54) : (0x1ULL << 62);
```